src/components/Search/FileUploader.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from 'react';
+import { Col, Upload, Button } from 'antd';
+import { UploadOutlined } from '@ant-design/icons';
+
+
+class FileUploader extends Component {
+
+    beforeUpload = file => {
+        const reader = new FileReader();
+        if (this.validateSize(file)) {
+            reader.onload = () => {
+                const json = JSON.parse(reader.result);
+                const geoid = file.name.replace('geo','').replace('.json','');
+
+                if (this.validateFile(json)) {
+                    this.props.onUpload(json, geoid);
+                }
+            };
+            reader.readAsText(file);
+
+            // Prevent upload
+            return false;
```

Comment:
I think your logic is not correct here.

In the case that the file is valid, this returns false and the comment says the uploaded is prevented.

Reply:
I think this is a weird ant design issue - if I remove the return false, it fails to upload. I think it's because this is on `beforeUpload`.

Reply:
That's weird. 

What if you return true? 


https://ant.design/components/upload/

Reply:
![Screen Shot 2020-06-05 at 11 01 50 AM](https://user-images.githubusercontent.com/16282020/83908471-08654980-a71c-11ea-97d9-7c508aecba9e.png)
 
If I return true here, the file upload fails.

Reply:
bizarre. carry on